linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -204,14 +204,27 @@ private[emitter] object CoreJSLib {
           } else {
             val Float32ArrayRef = globalRef("Float32Array")
 
+            /* (function(array) {
+             *   return function(v) {
+             *     array[0] = v;
+             *     return array[0];
+             *   }
+             * })(new Float32Array(1))
+             *
+             * Allocating the Float32Array once and for all, and capturing it
+             * in an IIFE, is *much* faster than recreating it in every call of
+             * the polyfill (about an order of magnitude).
+             */
             val array = varRef("array")
-            val typedArrayPolyfill = genArrowFunction(paramList(v), {
+            val typedArrayPolyfillInner = genArrowFunction(paramList(v), {
               Block(
-                  const(array, New(Float32ArrayRef, 1 :: Nil)),
                   BracketSelect(array, 0) := v,
                   Return(BracketSelect(array, 0))
               )
             })
+            val typedArrayPolyfill = Apply(
+                genArrowFunction(paramList(array), Return(typedArrayPolyfillInner)),
+                New(Float32ArrayRef, 1 :: Nil) :: Nil)
```

Comment:
FYI there is a `genIIFE` helper, but I'm not sure it's better to use it here... :shrug: Up to you.

Reply:
`genIIFE` is specialized for no-capture lambdas. But here we are actually creating a capture that we pass into the IIFE, so `genIIFE` is not applicable.

Reply:
Ah no I'm completely wrong. `genIIFE` does handle captures. But it expects `ParamDef`s to cater to the needs of `FunctionEmitter`, so it's still not very nice to use in this specific context.